store/driver/txn/unionstore_driver.go

```diff
@@ -19,41 +19,73 @@ import (
 
 	"github.com/pingcap/tidb/kv"
 	derr "github.com/pingcap/tidb/store/driver/error"
+	tikverr "github.com/tikv/client-go/v2/error"
 	tikvstore "github.com/tikv/client-go/v2/kv"
 	"github.com/tikv/client-go/v2/tikv"
+	"github.com/tikv/client-go/v2/util"
 )
 
+type visibilityChecker interface {
+	isVisible(kv.Key) bool
```

Comment:
The name is a bit vague. The status is 3-valued: (1)exist + visible, (2)exist + invisible, (3)absent(unknown). 
It returns true iff (1) || (3).

Reply:
How about to rename it to `invisible`?

Reply:
Sounds good. Or something like `isVisibleOrUnknown`(?)